### PR TITLE
[SPARK-12995][GraphX] Remove deprecate APIs from Pregel

### PR DESCRIPTION
**Pull request by maropu**

### CODE REVIEW

**Comment by maropu** [comment]
@srowen This is an activity from the discussion in #4402.
I checked that GraphX has deprecate APIs used only in Pregel and this pr removes them.
If there aren't any problems, I'll also remove deprecate ones from the test codes in GraphX.

**Comment by srowen** [comment]
While we're here we can remove runSVDPlusPlus in SVDPlusPlus?

File: graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala [hunk 1]
```diff
@@ -109,19 +109,17 @@ object Pregel extends Logging {
    * @return the resulting graph at the end of the computation
    *
    */
-  def apply[VD: ClassTag, ED: ClassTag, A: ClassTag]
-     (graph: Graph[VD, ED],
+  def apply[VD: ClassTag, ED: ClassTag, A: ClassTag](
```
Comment (srowen) [inline graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala]: Although it's really minor, the paren placement now makes it hard to see that there are two arg lists. I think it was better as-is.

**Comment by srowen** [comment]
Jenkins, retest this please

File: graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala (hunk 1 above)
Comment (maropu) [inline graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala]: Fixed.

**Comment by maropu** [comment]
Ah, yes. I removed runSVDPlusPlus in this pr.

**Comment by srowen** [comment]
@maropu yeah I think you'll need to write MiMa exclusions for the removal, to confirm they are intended.

**Comment by maropu** [comment]
yeah, I'm working on it.

**Comment by maropu** [comment]
@srowen Finished.

**Comment by srowen** [comment]
Looks OK to me, though I'lll want to leave it for review for a while. CC @ankurdave if possible

**Comment by maropu** [comment]
Okay

**Comment by maropu** [comment]
@ankurdave @srowen ping

**Comment by maropu** [comment]
@srowen @ankurdave almost inactive, so what should I do?

**Comment by srowen** [comment]
I'm out this week on vacation but definitely intend to proceed next week. I was leaving it for a while in case ankur is around

**Comment by maropu** [comment]
okay.

**Comment by srowen** [comment]
I'm back now. I'm almost ready to merge, but I noticed: should this not also remove `mapReduceTriplets`? Also the MiMa exclusions are correct but seem to be in the wrong place; at least, it might be clearer to make a new Seq section containing the exclusions just for this issue.

**Comment by maropu** [comment]
Welcome back. We cannot remove `mapReduceTripelets` because some tests in graphx still use it.
So, I'll make another pr to fix the tests, then remove it.

**Comment by maropu** [comment]
The position of the mima entry fixed.

**Comment by srowen** [comment]
If the only usage is tests, then the tests should be removed too, if they're just there to test this deprecated method. If it's found in other tests, can you migrate that code accordingly?

**Comment by maropu** [comment]
Ah, we cannot simply remove some tests because these tests target other tests for graph processing.
So, we need to replace `mapReduceTriplets` with newer `aggregateMessage`.
Yes, I can fix them.

**Comment by maropu** [comment]
Jenkins, retest this please.

**Comment by srowen** [comment]
@maropu I'm ready to merge this, but it needs a rebase, and I think this should be completed by removing `mapReduceTriplets` entirely

**Comment by maropu** [comment]
@srowen okay, I removed `mapReduceTriplets' in this pr.

**Comment by maropu** [comment]
@srowen okay, all the tests passed, so plz check this?

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**